### ISSUE TITLE
Can't listen on any expose server ports, local ports work fine

### ISSUE THREAD

**Issue by MarvelousMari** [beyondcode/expose#98]
Hey I'm a bit new to this but I'm having an issue. I can use postman to post to my local site and everything works fine but when I try to post to the provided domain at expose the server recognizes a post was attempted yet my script doesn't catch it. 

The ruby Script that's running (uses sinatra)

```ruby
require 'sinatra'
require 'json'
print "running Ruby Script\n"
# catches payload push and gets the JSON
#post '/payload' do
post '/'do
  print "in payload\n"
  push = JSON.parse(request.body.read)
  puts "I got some JSON: #{push.inspect}"
end
```

When I use postman to POST to localhost:4567 I get the following out of my ruby script:

```
marvelousmari@pop-os:~/Documents/test$ ruby myWebApp.rb 
running Ruby Script
/var/lib/gems/2.7.0/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1526: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/var/lib/gems/2.7.0/gems/rack-2.2.3/lib/rack/handler/thin.rb:11: warning: The called method `run' is defined here
== Sinatra (v2.0.8.1) has taken the stage on 4567 for development with backup from Thin
Thin web server (v1.7.2 codename Bachmanity)
Maximum connections set to 1024
Listening on localhost:4567, CTRL+C to stop
in payload
I got some JSON: { 

```
and then it shows the json I sent it.

When I send the same thing with Postman to https://webhooktest.sharedwithexpose.com/ I get a 502 Error from nginx about half the time the other half of the time I get nothing. It shows this from the CLI
```
marvelousmari@pop-os:~/Documents/test$ expose --subdomain=webhooktest
Thank you for using Expose!

⭐️ Support us by starring the repo at https://github.com/beyondcode/expose ⭐️
---
Local-URL:		test.test
Dashboard-URL:		http://127.0.0.1:4040
Expose-URL:		https://webhooktest.sharedwithexpose.com

+--------+-----+----------+---------------------+----------+
| Method | URI | Response | Time                | Duration |
+--------+-----+----------+---------------------+----------+
| POST   | /   |          | 2020-07-17 21:04:32 | 12ms     |
+--------+-----+----------+---------------------+----------+
```
When I look at it on the dashboard I can see the POST but I can't even click the response tab. I've run this exact same set up with ngork and It worked fine, but I'd much rather use expose. Please let me know if you need to post an log files or any more info!

**Comment by MarvelousMari** [comment]
I also can't get the settings for the server with a GET /api/settings or the sites being shared....

**Comment by keevitaja** [comment]
which OS are you using? i am trying to get expose working with free sharedwithexpose.com and no response. just hangs there idle.

**Comment by KCknows** [comment]
I wonder why??
Hahaha

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by MarvelousMari** [comment]
Weird lol

I use Pop!_OS , I changed my /etc/hosts file to 

```
127.0.0.1	localhost	
::1		localhost
127.0.1.1	pop-os.localdomain	pop-os
127.0.0.2	test.test
```

my expose server is running like this
```
marvelousmari@pop-os:~/Documents/test$ expose --subdomain=webhooktest test.test
Thank you for using Expose!

⭐️ Support us by starring the repo at https://github.com/beyondcode/expose ⭐️
---
Local-URL:		test.test
Dashboard-URL:		http://127.0.0.1:4040
Expose-URL:		https://webhooktest.sharedwithexpose.com

+--------+----------+---------------+---------------------+----------+
| Method | URI      | Response      | Time                | Duration |
+--------+----------+---------------+---------------------+----------+
| GET    | /        | 200 OK        | 2020-07-25 19:51:59 | 56ms     |
| POST   | /        | 200 OK        | 2020-07-25 19:50:57 | 55ms     |
| POST   | /payload | 404 Not Found | 2020-07-25 19:47:39 | 70ms     |
+--------+----------+---------------+---------------------+----------+

```

I can get proper responses now! Yet, my current directory isn't what's being shared... that GET request returns the Apache2 Ubuntu Default Page... I'm not sure if I need to whitelist the directory I'm trying to share with expose or how to do that. My ruby script looks like 

```
require 'sinatra'
require 'json'

set :bind, 'test.test'

print "running Ruby Script\n"
# catches payload push and gets the JSON
post '/payload' do
#post '/' do
  print "in payload\n"
  push = JSON.parse(request.body.read)
  puts "I got some JSON: #{push.inspect}"
end

# TODO: need to add SECRET to secure webhook
```
Posts to https://webhooktest.sharedwithexpose.com/payload return a 404 error but test.test:4567/payload is caught just fine by  my ruby script... 

The 404 response includes this in the 
```
Server at test.test Port 80
```
So I tried setting the port in my ruby script and kept getting In Use Errors.

Not sure what to do from here.. Any advice?

**Comment by MarvelousMari** [comment]
...Wow okay I changed  a few things

I specified the tld in the config.php file, and then called to start the server this way.
```
expose share test.expose:4567 --subdomain=webhooktest
```

So to solve this problem completely, I added to my `/etc/hosts` an address and the name I specified as the tld in the config.php file. Then I launched the server explicitly using the tld and port number and it's working fine now!
